src/track/cueinfo.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+// cueinfo.h
+// Created 2020-02-28 by Jan Holthuis
+
+#include "util/color/color.h"
+
+// DTO for Cue information without dependencies on the actual Track object
+class CueInfo : public QObject {
+    Q_OBJECT
+
+  public:
+    enum class Type {
+        Invalid = 0,
+        HotCue = 1,
+        MainCue = 2,
+        Beat = 3, // unused (what is this for?)
+        Loop = 4,
+        Jump = 5,
+        Intro = 6,
+        Outro = 7,
+        AudibleSound = 8, // range that covers beginning and end of audible
+                          // sound; not shown to user
+    };
+
+    static constexpr double kNoPosition = -1.0;
+    static const int kNoHotCue = -1;
+
+    CueInfo();
+    CueInfo(CueInfo::Type type,
+            double startPositionMilliseconds,
```

Comment:
Is `double` the right type? I don't know if we need fractions of milliseconds. Serato just uses `quint32`.

Reply:
Considering sample rates of >= 44.1 kHz then 1000 positions in a second would not be precise enough to preserve the actual sample position due to rounding errors. Not really needed for playback, but more to avoid precision loss during conversions.